pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+  EvictionMinimumReclaim string `json:"evictionMinimumReclaim"`
+  // If enabled, the kubelet will integrate with the kernel memcg notification to determine if memory eviction thresholds are crossed rather than polling.
+  ExperimentalKernelMemcgNotification *bool `json:"experimentalKernelMemcgNotification"`
+}
+
+// Volume management
+type VolumeConfig struct {
+  // How frequently to calculate and cache volume disk usage for all pods
+  VolumeStatsAggPeriod metav1.Duration `json:"volumeStatsAggPeriod"`
+  // volumePluginDir is the full path of the directory in which to search
+  // for additional third party volume plugins
+  VolumePluginDir string `json:"volumePluginDir"`
+  // enableControllerAttachDetach enables the Attach/Detach controller to
+  // manage attachment/detachment of volumes scheduled to this node, and
+  // disables kubelet from executing any attach/detach operations
+  EnableControllerAttachDetach *bool `json:"enableControllerAttachDetach"`
+}
+
+// Node validation and setup
+type NodeConformanceConfig struct {
+  // TODO(#34726:1.8.0): Remove the opt-in for failing when swap is enabled.
+  // Tells the Kubelet to fail to start if swap is enabled on the node.
+  ExperimentalFailSwapOn bool `json:"experimentalFailSwapOn,omitempty"`
+  // This flag, if set, enables a check prior to mount operations to verify that the required components
+  // (binaries, etc.) to mount the volume are available on the underlying node. If the check is enabled
+  // and fails the mount operation fails.
+  ExperimentalCheckNodeCapabilitiesBeforeMount bool `json:"experimentalCheckNodeCapabilitiesBeforeMount,omitempty"`
+  // Default behaviour for kernel tuning
+  ProtectKernelDefaults bool `json:"protectKernelDefaults"` // If true, Kubelet will error if kernel parameters aren't what it expects. If false, Kubelet will try to set kernel parameters to what it wants.
+}
+
+// Events
+type EventConfig struct {
+  // eventRecordQPS is the maximum event creations per second. If 0, there
+  // is no limit enforced.
+  EventRecordQPS *int32 `json:"eventRecordQPS"`
+  // eventBurst is the maximum size of a bursty event records, temporarily
+  // allows event records to burst to this number, while still not exceeding
+  // event-qps. Only used if eventQps > 0
+  EventBurst int32 `json:"eventBurst"`
+}
+
+// Cloud provider configuration
+type CloudConfig struct {
+ // cloudProvider is the provider for cloud services.
+  CloudProvider string `json:"cloudProvider"`
+  // cloudConfigFile is the path to the cloud provider configuration file.
+  CloudConfigFile string `json:"cloudConfigFile"`
+}
+
+// Debug settings
+type KubeletDebugConfig struct {
+  // enableDebuggingHandlers enables server endpoints for log collection
+  // and local running of containers and commands
+  EnableDebuggingHandlers *bool `json:"enableDebuggingHandlers"`
```

Comment:
Caveat: `EnableDebuggingHandlers` is enabled by default because it's required for exec, attach, logs, portforward, etc to work. It's probably a misnomer, and I am not sure it should really be grouped into the `KubeletDebugConfig`.